### PR TITLE
Feature/feature/104 english localization

### PR DESCRIPTION
**Pull request by ICTEnablers**
I've added localization to english using the react-i18next library.  Only issue is that it uses hooks for all functionality and hooks need a react component to work. So this  does not work in fingerPlacement and other "helper" functions that area not part of a component.

I implemented a work around for it now, will add an issue about it when this is merged.

### CODE REVIEW

File: src/app.tsx [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
 import Explore from './components/explore';
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 import Task from './components/task';
 import Summary from './components/summary/summary';
 import Home from './components/home/home';
 
-const App = () => (
-  <Router>
+const Start = () => {
+  return (<Router>
```
Comment (duff92) [inline src/app.tsx]: Redundant to have a return statement here. See how it was before :)

File: src/components/explore/explore-menu.tsx [hunk 2]
```diff
@@ -76,19 +80,34 @@ const ExploreMenu = (props: IProps) => {
...
+
   useEffect(() => {
     if (menuElement && menuElement.current) {
       menuElement.current.focus();
     }
   }, []);
 
   useEffect(() => {
-    speak(headerText + ' ' + introText).then(url => playAudio(audioElementIntro, url));
+    const ttsOptionsInEffect: ITTS = { language: i18n.language };
```
Comment (duff92) [inline src/components/explore/explore-menu.tsx]: Why do you have `ttsOptionsInEffect` and `ttsOptions` on row 83?

File: src/components/explore/explore.tsx [hunk 3]
```diff
@@ -71,24 +75,28 @@ const Explore = props => {
...
 
   useEffect(() => {
     let interval;
 
     if (timeCount > timeForExercise || explore.typeCount > maxInputs) {
-      setHeaderText('Redo');
-      setIntroText(`Bra jobbat! ${currentGameCharacter.name} har fått ett rött bälte och är redo för sitt första uppdrag, tryck enter för att starta.`);
-      completedAction();
+      setHeaderText(t('explore.completedHeader'));
+      setIntroText(t('explore.wellDone') + ' ' + currentGameCharacter.name+ ' ' + t('explore.readyText'));
```
Comment (duff92) [inline src/components/explore/explore.tsx]: Missing space `+ currentGameCharacter.name +`

File: src/components/explore/explore.tsx [hunk 4]
```diff
@@ -71,24 +75,28 @@ const Explore = props => {
...
+      setHeaderText(t('explore.completedHeader'));
+      setIntroText(t('explore.wellDone') + ' ' + currentGameCharacter.name+ ' ' + t('explore.readyText')); 
+      completedAction();  
     } else {
       interval = setInterval(() => setTimeCount(0), 1000);
     }
 
     return () => clearInterval(interval);
-
+    //ignore lint warnigns about t(...)
```
Comment (duff92) [inline src/components/explore/explore.tsx]: Add space in comment

File: src/components/layout/InstructionLayout.tsx [hunk 5]
```diff
@@ -42,14 +43,19 @@ const InstructionLayout = (props: IProps) => {
...
     if (linkElement.current) {
       linkElement.current.focus();
     }
   }, []);
 
   useEffect(() => {
-    if (instructionToSpeak !== undefined) speak(instructionToSpeak).then(url => playAudio(audioInstructionElement, url));
+    const ttsOptionsInEffecct: ITTS = { language: i18n.language }
+    if (instructionToSpeak !== undefined) speak(instructionToSpeak, ttsOptionsInEffecct).then(url => playAudio(audioInstructionElement, url));
+    //ignore lint i18n warning
```
Comment (duff92) [inline src/components/layout/InstructionLayout.tsx]: Add space in comment `// ignore...`

File: src/components/summary/summary.tsx [hunk 6]
```diff
@@ -26,24 +27,29 @@ type ISummmaryProps = StateProps;
...
   useEffect(() => {
+    const ttsOptionsInEffect: ITTS = { language: i18n.language };
     playAudio(audioElement, assetBaseUrl + 'done.mp3').then(() => {
-      speak(feedbackText).then(url => {
+      speak(t('summary.completedText'), ttsOptionsInEffect).then(url => {
         playAudio(audioElement, url).catch(error => console.error('playAudio error', error));
       }).catch(error => console.error('speak error', error));
     }).catch(error => console.error('play audio error', error));
-  }, [feedbackText]);
+    //ignore lint i18n warning
```
Comment (duff92) [inline src/components/summary/summary.tsx]: Add space in comment

File: src/components/task/task-countdown.tsx [hunk 7]
```diff
@@ -45,8 +49,11 @@ const TaskCountDown = (props: IProps) => {
   }, [timeLeft, to, props.history]);
 
   useEffect(() => {
+    const ttsOptionsInEffect: ITTS = { language: i18n.language };
     // todo: Fix to start counting at starttime and not starttime - 1
-    speak(timeLeft + ' ').then(url => playAudio(audioElement, url));
+    speak(timeLeft + ' ', ttsOptionsInEffect).then(url => playAudio(audioElement, url));
+    //ignore lint i18n warning
```
Comment (duff92) [inline src/components/task/task-countdown.tsx]: Add space in comment

File: src/components/task/task.tsx [hunk 8]
```diff
@@ -101,10 +103,13 @@ const Task = props => {
   }, [currentPos, task.exercise, ttsOptions]);
 
   useEffect(() => {
-    speak(task.exercise[currentPos].text, ttsOptions).then(url => 
+    const ttsOptionsInEffect: ITTS = { language: i18n.language, rate: 2 };
+    speak(task.exercise[currentPos].text, ttsOptionsInEffect).then(url => 
       playAudio(audioElement, url).catch(error => 
         console.error('play error intial character ' + error))).catch(error => 
           console.error('speak inital character errror ' + error));
+      //Ignore lint warning about currentPos, i18n and audioElement
```
Comment (duff92) [inline src/components/task/task.tsx]: Add space in comment

File: src/app.tsx (hunk 1 above)
Comment (ICTEnablers) [reply src/app.tsx]: Removing it yields the error:
JSX element type 'void' is not a constructor function for JSX elements.
at line 28

File: src/components/explore/explore-menu.tsx (hunk 2 above)
Comment (ICTEnablers) [reply src/components/explore/explore-menu.tsx]: If i used the externally declared const ttsOptions I get a dependency warning for useEffect, that is why I declared a new const ttsOptionsInEffect in side useEffect

File: src/components/explore/explore.tsx (hunk 4 above)
Comment (ICTEnablers) [reply src/components/explore/explore.tsx]: I fixed the typo, where should the space go exactly?

Comment (ICTEnablers) [reply src/components/explore/explore.tsx]: nevermind saw it now, fixed

File: src/components/layout/InstructionLayout.tsx (hunk 5 above)
Comment (ICTEnablers) [reply src/components/layout/InstructionLayout.tsx]: fixed

File: src/components/task/task-countdown.tsx (hunk 7 above)
Comment (ICTEnablers) [reply src/components/task/task-countdown.tsx]: fixed

File: src/components/summary/summary.tsx (hunk 6 above)
Comment (ICTEnablers) [reply src/components/summary/summary.tsx]: fixed

File: src/components/explore/explore.tsx (hunk 3 above)
Comment (ICTEnablers) [reply src/components/explore/explore.tsx]: fixed

**Pull request merged**